python/pyspark/sql/tests/test_utils.py

```diff
@@ -739,33 +736,146 @@ def test_assert_unequal_null_expected(self):
             exception=pe.exception,
             error_class="INVALID_TYPE_DF_EQUALITY_ARG",
             message_parameters={
-                "expected_type": Union[DataFrame, ps.DataFrame, List[Row]],
+                "expected_type": "Union[DataFrame, ps.DataFrame, List[Row]]",
                 "arg_name": "expected",
                 "actual_type": None,
             },
         )
 
     def test_assert_equal_exact_pandas_df(self):
+        import pandas as pd
```

Comment:
can you conditionally skip this? e.g., see `pyspark.testing.sqlutils.have_pandas`

Reply:
Maybe we should also consider for `numpy` as well? Please check `pyspark.testing.utils.have_numpy`.